marray/tests/test_marray.py

```diff
@@ -196,9 +194,10 @@ def test_rarithmetic_binary(f, seed=None):
     ref = np.ma.masked_array(ref_data, mask=ref_mask)
     assert_equal(res, ref, seed)
 
-    # res = f(marrays[1].data, marrays[0])
-    # ref = f(masked_arrays[1].data, masked_arrays[0])
-    # assert_equal(res, ref, seed)
+    # Check that reflected operator works with Python scalar
+    res = f(2, marrays[0])
+    ref = f(2, masked_arrays[0])
+    assert_equal(res, ref, seed)
```

Comment:
I wanted to see if the reflected operations would work with NumPy array + Masked Array, but I don't know how to get that working. Here's a demonstration of why:

```python3
import numpy as xp

class MyClass:
    def __init__(self, x):
        self.data = x

    def __radd__(self, other):
        print(other)
        return MyClass(self.data + other)

x = xp.arange(5)
y = MyClass(x)
x + y
```
Produces
```
0
1
2
3
4
Out[2]: 
array([<__main__.MyClass object at 0x000001F5F2177AA0>,
       <__main__.MyClass object at 0x000001F5F2177A10>,
       <__main__.MyClass object at 0x000001F5F22158B0>,
       <__main__.MyClass object at 0x000001F5F2215850>,
       <__main__.MyClass object at 0x000001F5F22141A0>], dtype=object)
```
I guess when NumPy's `__add__` method fails, NumPy passes each element of the array to our `__radd__` method individually rather than passing the whole array.

Strict Array + Masked Array doesn't do this; it produces the correct result.

In the meantime, I replaced the array with a scalar. Picking a particular value manually (`2`) doesn't really fit with my testing philosophy, but I will adjust that later. Draft quickly, polish later.

Reply:
same as the other comment: I'll try to see if I can spot anything tomorrow, otherwise agreed, let's come back to this sometime later.